### DISCUSSION TITLE
Reading data across non-child components

### DISCUSSION THREAD

**Discussion by raffle-x-dev** [preactjs/preact#4300] [Q&A] [answered]
Is there a way to read a variable created inside a component, from another non-child component? I thought it could have been done with signals, but I might have misunderstood it.

**Comment by marvinhagemeister** [comment, marked as answer]
Yes, pass a signal to the components that should read or write to it and update it.

**Reply by raffle-x-dev** [reply]
Aw... That's exactly what I'd like to do. No idea how to pass a signal to siblings components. I guess it's my limit. I've seen workarounds to pass variables to parent components but they were dozens of lines of code just to kick out a variable. I got scared to implement that, was looking for something easier. 😅

**Reply by rschristian** [reply]
Signals are pretty neat as you can just import them.

```js
// siblingOne.js
export const myCount = signal(0);

export function SiblingOne() { ... }
```

```js
// siblingTwo.js
import { myCount as mySiblingCount } from './siblingOne.js'

export function SiblingTwo() { ... }
```

Of course, lifting state up into the parent is also a viable strategy. Parent can pass the signal to both children, allowing them to read/write the same shared instance.


```js
const sharedCount = signal(0);

export function ParentComponent() {
    return (
        <>
            <ChildOne count={sharedCount} />
            <ChildTwo count={sharedCount} />
        </>
    );
}
```

**Reply by raffle-x-dev** [reply]
Thanks for taking time to answer this.
I guess my problem is that my signals are created inside the component logic, which uses props:
```
// siblingTwo.js

export function SiblingTwo(props) {
   
    ...some logic using props...

    const mySignal = signal(uses above logic)

    return(...)
}
```

So you can't just export `mySignal` from inside the component function.
I partially solved by writing the value I need to localStorage.
If anyone has a better alternative or signals can help differently, I'd be more than happy to learn! @marvinhagemeister ✌

**Reply by rschristian** [reply]
Sounds like a good candidate for being raised into the parent and passed into the children then.
